### PR TITLE
[RFR] Dockerized the project

### PR DESCRIPTION
**Pull request by jaljo**
Dockerized the project so it cand be deployed to a DigitalOcean droplet

### CODE REVIEW

File: .docker/app/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM node:11.10.0-alpine as prod
+
+WORKDIR /app
+RUN chown 1000:1000 /app
+
+USER 1000
+
+CMD ["yarn"]
```
Comment (nicolasmure) [inline .docker/app/Dockerfile]: please remove this line

File: .docker/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+version: '3.4'
+
+services:
+  app:
+    image: i24-reader:app-dev
```
Comment (nicolasmure) [inline .docker/docker-compose.yml]: please set proper image name

File: .docker/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    build:
+      context: ../
+      dockerfile: .docker/app/Dockerfile
+      target: prod
+    command: 'yarn start'
+    tty: true
```
Comment (nicolasmure) [inline .docker/docker-compose.yml]: you can also remove this line.

File: server.js [hunk 4]
```diff
@@ -14,6 +14,6 @@ app.get("*", function (request, response) {
 });
 
 // listen for requests :)
-var listener = app.listen(process.env.PORT, function () {
+var listener = app.listen(3000, function () {
```
Comment (nicolasmure) [inline server.js]: could be nice to keep the env var

File: .docker/app/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+COPY --chown=1000:1000 package.json package.json
+COPY --chown=1000:1000 yarn.lock yarn.lock
+
+RUN yarn install
+
+RUN yarn run build
+
+CMD ["yarn", "run", "server"]
```
Comment (nicolasmure) [inline .docker/app/Dockerfile]: there are no `server` script in your package.json file
perhaps you meant `start` ?

File: .docker/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+version: '3.4'
+
+services:
+  app:
+    image: i24-reader:app-dev
+    build:
+      context: ../
+      dockerfile: .docker/app/Dockerfile
+      target: prod
+    command: 'yarn start'
```
Comment (nicolasmure) [inline .docker/docker-compose.yml]: no need to set the command here if it does not differ from the one of the Dockerfile.

File: .docker/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+version: '3.4'
```
Comment (nicolasmure) [inline .docker/docker-compose.yml]: Looks like this compose file is used for dev env, so better rename it `docker-compose.dev.yml`.

File: .docker/docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+services:
+  app:
+    image: i24-reader:app-dev
+    build:
+      context: ../
+      dockerfile: .docker/app/Dockerfile
+      target: prod
```
Comment (nicolasmure) [inline .docker/docker-compose.yml]: why to specify the target when you only have one target in the Dockerfile ? (ie no multistage builds)

File: .docker/app/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,17 @@
+FROM node:11.10.0-alpine as prod
```
Comment (nicolasmure) [inline .docker/app/Dockerfile]: IMHO you can remove the target name as there is only one stage in this dockerfile.
Also, what about using a more recent version of node ?

File: .docker/app/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+WORKDIR /app
+RUN chown 1000:1000 /app
+
+USER 1000
+
+CMD ["yarn"]
+
+COPY --chown=1000:1000 package.json package.json
+COPY --chown=1000:1000 yarn.lock yarn.lock
```
Comment (nicolasmure) [inline .docker/app/Dockerfile]: you should also copy the sources ;)

File: .docker/docker-compose.yml (hunk 7 above)
Comment (nicolasmure) [reply .docker/docker-compose.yml]: You can define a `COMPOSE_FILE=.docker/docker-compose.dev.yml` env var in the `.env.dist`  file, which will be used to specify which compose file to use.

Comment (jaljo) [reply .docker/docker-compose.yml]: Isn't that compose file needed for production too ? :thinking:

Comment (nicolasmure) [reply .docker/docker-compose.yml]: then define a prod compose file : `docker-compose.prod.yml` ;)

File: .docker/app/Dockerfile (hunk 10 above)
Comment (jaljo) [reply .docker/app/Dockerfile]: :duck:

File: .docker/app/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+RUN chown 1000:1000 /app
+
+USER 1000
+
+COPY --chown=1000:1000 package.json package.json
+COPY --chown=1000:1000 yarn.lock yarn.lock
+
+RUN yarn install
+
+COPY --chown=1000:1000 .env.dist .env
```
Comment (nicolasmure) [inline .docker/app/Dockerfile]: you can use a single COPY : `COPY --chown=1000:1000 . /app`
and declare a `.dockerignore` file where you allow the files/dirs that should be copied ;)

File: .docker/docker-compose-dev.yml [hunk 12]
```diff
@@ -0,0 +1,15 @@
+version: '3.4'
```
Comment (nicolasmure) [inline .docker/docker-compose-dev.yml]: please rename this file `docker-compose.dev.yml` (no dash before `dev`)

**Review by nicolasmure — COMMENTED** [review]
How do you plan to deploy and run the app in prod ?

File: .docker/app/Dockerfile (hunk 11 above)
Comment (jaljo) [reply .docker/app/Dockerfile]: Thanks for that advice, cleaner indeed !

**Comment by jaljo** [comment]
Like discussed at Hackathon, you advise me to deploy it to a DigitalOcean droplet. Can we take some time to discuss about it ?

**Comment by jaljo** [comment]
Closed, as the whole application will be served over nginx rather than express

**Pull request closed without merging**